### PR TITLE
CHROM-194 Add All Number Format Subcomponents to Storybook

### PR DESCRIPTION
**Pull request by MMFane**
# What Was Changed

- Phone number was already in there - added Unit, Percent, and Price
- Grouped them all under "Number Format Field"
- Added all unique props to each story
- Referenced and linked to the components each extend or implement
- Had to do some rebasing (_one day I will remember to prefix Chroma commits with \<tag\>:_ ). Let me know if anything looks funky.

# Screenshots

## Before
Screen Shot 2022-08-03 at 11 44 59 AM

## After
Screen Shot 2022-08-03 at 11 45 12 AM
Screen Shot 2022-08-03 at 11 45 26 AM

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -71,8 +71,7 @@ export const useStyles = makeStyles(
         boxShadow: theme.boxShadows.focusVisibleInverse,
       },
       '&:disabled, &[disabled]': {
-        backgroundColor: 'rgba(255, 255, 255, 0.45)',
-        opacity: 1,
+        backgroundColor: theme.hexToRgba(theme.palette.common.white, 0.45),
```
Comment (xdumaine) [inline src/components/Button/Button.tsx:74]: Were these intentional in this PR?

**Comment by HaleyWardo** [comment]
Yeah it looks like your rebase didn't work as expected. I am seeing a bunch of `rgba` changes.

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (MMFane) [reply src/components/Button/Button.tsx:74]: Hmm... no, I don't think so. That may be an artifact of my rebase. Looking into it

**Comment by MMFane** [comment]
> Yeah it looks like your rebase didn't work as expected. I am seeing a bunch of `rgba` changes.

Damn, no kidding! Let's try that again

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (MMFane) [reply src/components/Button/Button.tsx]: @xdumaine @HaleyWardo OK, fixed it!

**Review by xdumaine — APPROVED** [review]

**Pull request merged**